python/cudf/cudf/_lib/pylibcudf/types.pxd

```diff
@@ -0,0 +1,63 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.
+
+from libc.stdint cimport int32_t
+from libcpp cimport bool as cbool
+
+from cudf._lib.cpp.types cimport data_type, interpolation, null_policy, type_id
+
+ctypedef int32_t underlying_type_t_type_id
+cpdef enum TypeId:
+    EMPTY = <underlying_type_t_type_id> type_id.EMPTY
```

Comment:
It does seem like we're fighting cython here rather.

Reply:
Yes, although I believe this is fixed in Cython 3.